server/api/router.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import express from 'express';
+import { createProxyServer } from 'http-proxy';
+import { UIServerModule } from 'types';
+
+import {
+  proxyErrorHandler,
+  proxyCompleteHandler,
+  proxyStartHandler,
+} from './controller';
+
+const moduleName = 'api';
+
+export const ApiModule: UIServerModule = {
+  moduleName,
+  addModule: (logGenerator, authFn, serverConfig) => {
```

Comment:
I see the logging staying as is here, but how it works underneath changing (I expect it will delegate to pino). The `logGenerator` here is the singleton - it has an entry and 'event' handler, where entry returns exit and debug etc. It has module name etc baked in. The logger here is meant to log the `addModule` function only (the same logger is in `res.locals`, and is meant to be used by callbacks)

Reply:
Not entirely comfortable with having to pass the log generator into a function but can revisit as this matures